packages/flame_audio/lib/bgm.dart

```diff
@@ -3,19 +3,28 @@ import 'dart:io';
 import 'package:audioplayers/audioplayers.dart';
 import 'package:flutter/widgets.dart';
 
+/// {@template _bgm}
 /// The looping background music class.
 ///
 /// This class helps with looping background music management that reacts to
 /// application lifecycle state changes. On construction, the instance is added
 /// as an observer to the [WidgetsBinding] instance. A [dispose] function is
 /// provided in case this functionality needs to be unloaded but the app needs
 /// to keep running.
+/// {@endtemplate}
 class Bgm extends WidgetsBindingObserver {
   bool _isRegistered = false;
+
+  /// [AudioCache] instance of the [Bgm].
   late AudioCache audioCache;
+
+  /// The [AudioPlayer] instance that is currently playing the audio.
   AudioPlayer? audioPlayer;
+
+  /// If the [Bgm] is playing or not.
```

Comment:
my most misspelled word :P
```suggestion
  /// Whether [Bgm] is playing or not.
```